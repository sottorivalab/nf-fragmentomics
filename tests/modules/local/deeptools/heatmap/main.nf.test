nextflow_process {

    name "Test Process HEATMAP"
    script "modules/local/deeptools/heatmap/main.nf"
    process "HEATMAP"

    test("Should run HEATMAP without failures") {

        when {
            params {
                outdir = "tests/results"
            }
            process {
                // meta_sample [caseid, sampleid, timepoint], meta_target [source, name], matrix

                """
                input[0] = [
                    [ caseid:'test', sampleid: 'test', timepoint: 'T1' ],
                    [ source: 'test', name: 'test' ],
                    file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/deeptools/test.computeMatrix.mat.gz', checkIfExists: true)
                ]
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out.heatmap, process.out.versions).match()
        }
    }
}
