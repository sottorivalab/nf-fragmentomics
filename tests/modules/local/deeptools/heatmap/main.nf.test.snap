{
    "Should run HEATMAP without failures": {
        "content": [
            [
                [
                    {
                        "caseid": "test",
                        "sampleid": "test",
                        "timepoint": "T1"
                    },
                    {
                        "source": "test",
                        "name": "test"
                    },
                    "test.computeMatrix.mat_heatmap.png:md5,380bdf1e95fe94831c6b56ea298519bf"
                ]
            ],
            [
                "versions.yml:md5,f955ba6259d623b1744b53fd6ec85eed"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-01-16T14:43:25.048554"
    }
}